src/callback_handler/oci.rs

```diff
@@ -30,6 +31,17 @@ impl Client {
         serde_json::to_string(&image_digest)
             .map_err(|e| anyhow!("Cannot serialize response to json: {}", e))
     }
+
+    pub async fn manifest(&self, image: &str) -> Result<OciManifest> {
```

Comment:
is it ok to return the OciManifest type? shouldn't we be returning a String with the serialized json?. I suppose it happens automatically anywas once we send the callback  back. But then I wonder why for `digest()` we do the conversion to String.

Reply:
Yeah, I think it would be better to serialize that to JSON and then just return the JSON string

Reply:
I'm not sure if this is necessary. The digest is a string because the digest will be store inside the `ManifestDigestResponse` type (see `get_oci_digest_cached`) wich will be parsed into a [JSON string later](https://github.com/kubewarden/policy-evaluator/blob/26c4ec2143542c10293dab0c0c0b11716fe51a2c/src/callback_handler.rs#L45) on. Therefore, I do not see the reason of serializing the content here. Furthermore, if we return string, the response will be a string where the content is a serialized JSON. And not a serialized JSON value. Because we serialized it twice. Which means that the other end will need to serialize twice as well.  Once to get the string and the other to transform the string into a JSON value. 

Does it make sense? Am I missing something?

Reply:
What have I noticed is that the function is using a type from oci-distribution. And the rust SDK does not have it as a dependency. Because it does not import policy-fetcher and policy-evaluator. Therefore, we need to add this edge in  dependency graph (maybe using a transitive dependency). I remember that we change the dependency orientation some time ago for some reason. I guess is to simplify the dependency bumps. Another option is doing something similar to what the digest does and create a type in the SDk and use it here as well.

Reply:
Ok, scratch that. The host is going to serialize the `OciManifest` into JSON, and return that to the guest.

In terms of the SDKs, I would say it's up to us to decide how to handle that. The Rust SDK can provide a `get_manifest` a function that:

* Calls the host function
* Handles the error
* deserializes the JSON object into a Rust type
* return the Rust type

I think we should avoid to use the oci-distribution crate, since it's too fat for what we need. We should instead use the [`oci-spec`](https://crates.io/crates/oci-spec/) one, which contains only the definitions of the objects. In the beginning I would just introduce this extra dependency. The rust compiler should be smart enough and not add the library to the final `.wasm` file unless the code is actually used.

As for the Go one, we could do the very same OR we could just have the SDK return a generic [`RawMessage`](https://pkg.go.dev/encoding/json#RawMessage).
I would prefer to have a well defined Go Type to be returned, but we will have to find something that 1) builds with TinyGo 2) doesn't make all the policies bigger (especially the ones that are not using this capability)

These are however topics that need to be discussed when looking at the actual Rust/Go SDK.

This is a small feature, but it involves different projects. The way we tackled that in the past was to make PRs using code from PR branches. Go ahead and build a Rust policy that makes use of this capability, using a Rust SDK from a feature branch and a kwctl that consumes policy-evaluator from this PR branch

Reply:
I've opened an [PR](https://github.com/kubewarden/policy-sdk-rust/pull/129) adding the capability in the Rust SDK using the `oci-spec`. Take a look on it as well.

Reply:
@flavio @viccuad,  can we consider this PR fine? As far as I can see we can keep things as they are. Right?

Reply:
Yes, I think this is fine. However let's not merge it yet, at least not until we have tested kwctl with a policy written in Go and Rust that make use of this new capability